charts/k8s-service/templates/dbmigrate.yaml

```diff
@@ -0,0 +1,36 @@
+{{/* Go Templates do not support variable updating, so we simulate it using dictionaries */}}
+{{- $hasInjectionTypes := dict "hasVolume" false "hasEnvVars" false -}}
+{{- if .Values.envVars -}}
+  {{- $_ := set $hasInjectionTypes "hasEnvVars" true -}}
+{{- end -}}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: "{{ .Values.applicationName }}-db-migrate"
+  annotations:
+    "helm.sh/hook": pre-install,pre-upgrade
+    "helm.sh/hook-delete-policy": hook-succeeded
+spec:
+  activeDeadlineSeconds: 300
+  backoffLimit: 1
+  template:
+    name: "{{ .Values.applicationName }}-db-migrate"
+    spec:
+      restartPolicy: Never
+      containers:
+        - name: "{{ .Values.applicationName }}-db-migrate"
+          {{- $repo := required "containerImage.repository is required" .Values.containerImage.repository }}
+          {{- $tag := required "containerImage.tag is required" .Values.containerImage.tag }}
+          image: "{{ $repo }}:{{ $tag }}"
+          {{- if .Values.hookCommand }}
```

Comment:
Should we name this `migrateCommand` instead? We _could_ conceivably have other hooks in the future.

Reply:
Right, that makes sense. Initially, I thought of this hook as a more generic one but then yes, it became a migrator specifically.

Reply:
Updated